websites/R/RTLplay/presence.ts

```diff
@@ -105,20 +129,33 @@ function exist(selector: string) {
 	return document.querySelector(selector) !== null;
 }
 
-const shortenedURLs: Record<string, string> = {};
-async function getShortURL(url: string) {
-	if (!url || url.length < 256) return url;
-	if (shortenedURLs[url]) return shortenedURLs[url];
-	try {
-		const pdURL = await (
-			await fetch(`https://pd.premid.app/create/${url}`)
-		).text();
-		shortenedURLs[url] = pdURL;
-		return pdURL;
-	} catch (err) {
-		presence.error(err);
-		return url;
+function timeStringToMinutes(timeString: string) {
+	const timeParts = timeString.split(":").map(Number);
+	let totalSeconds = 0;
+
+	// Handle different lengths of time parts
+	switch (timeParts.length) {
+		case 4: // DD:HH:MM:SS
+			totalSeconds += timeParts[0] * 24 * 3600;
+			totalSeconds += timeParts[1] * 3600;
+			totalSeconds += timeParts[2] * 60;
+			totalSeconds += timeParts[3];
+			break;
+		case 3: // HH:MM:SS
+			totalSeconds += timeParts[0] * 3600;
+			totalSeconds += timeParts[1] * 60;
+			totalSeconds += timeParts[2];
+			break;
+		case 2: // MM:SS
+			totalSeconds += timeParts[0] * 60;
+			totalSeconds += timeParts[1];
+			break;
+		case 1: // SS
+			totalSeconds += timeParts[0];
+			break;
 	}
+
+	return Math.round(totalSeconds / 60);
 }
```

Comment:
I think there is a function for this in the presence class? Take a look at `presence.timestampFromFormat`

Reply:
Thanks, I've just tried it. I hadn't realized that it gave a format in seconds, I can convert it to minutes like the initial function. :)